radio/src/lua/interface.cpp

```diff
@@ -618,17 +618,17 @@ static bool luaLoad(const char * filename, ScriptInternalData & sid)
 #if defined(LUA_MODEL_SCRIPTS)
 static bool luaLoadMixScript(uint8_t ref)
 {
+    assert(ref >= SCRIPT_MIX_FIRST);    
```

Comment:
Well, I had the hope that in this code-base the assert-macros does something meaningful (e.g. triggeres EM, ...).
Most code-bases I work on use `assert()` not only to check real pre/post-conditions, but also to give advice to the maintenance programmer like a comment.
What's your advice here? 
The assert vanishes in non-debug compiles anyway, so it should crash the radio only for developers.
Should I trigger EM instead? How?

Reply:
```C++
if (ref < SCRIPT_MIX_FIRST) return false;
```

Reply:
Ok, made if conditional compile under DEBUG

Reply:
What was the purpose of wrapping that in DEBUG? Isn't that a input sanity check which is also valid on the radio?

Reply:
Well, maybe I'm totally wrong with this. 
Normally here an assert would go because the should assert the precondition for the following code to work correct.
@raphaelcoeffic advised me not to use assert here because it crashes the radio without a possibility to check where this crash happens as a kind of post-mortem-debug (like a core-file analysis on posix-systems).
So I changed it in the that simple `if`. And I made it active only under `DEBUG` because this should somehow resemble the `assert`-macro.

My remaining question is:
1) how to check pre- and postconditions as well as class-invariants in this code-base
2) what should happen if  the abose conditions aren't fullfilled? 
2a) Try to write some mark into a crash-dump-file? 
2b) In `DEBUG` or `RELEASE` build?
2c) Trigger 'EM'?
2d) ...

2